File: src/main/java/org/apache/prepbuddy/datacleansers/ImputationStrategy.java
package org.apache.prepbuddy.datacleansers;

import org.apache.prepbuddy.utils.RowRecord;

public interface ImputationStrategy {
    MissingDataHandler REMOVE_ROWS = new MissingDataHandler() {
        @Override
        public String handleMissingData(RowRecord record) {
            return null;
        }
    };

    MissingDataHandler SUSBTITUTE_WITH_MEAN = new MissingDataHandler() {
        @Override
        public String handleMissingData(RowRecord record) {
            return null;
        }
    };
    MissingDataHandler SUSBTITUTE_WITH_MOST_FREQUENT_ITEM = new MissingDataHandler() {
        @Override
        public String handleMissingData(RowRecord record) {
            return null;
        }
    };
}

File: src/main/scala/org/apache/datacommons/prepbuddy/cleansers/imputation/ImputationStrategy.scala
package org.apache.datacommons.prepbuddy.cleansers.imputation

import org.apache.datacommons.prepbuddy.rdds.TransformableRDD
import org.apache.datacommons.prepbuddy.utils.RowRecord

trait ImputationStrategy extends Serializable{
  def prepareSubstitute(rdd: TransformableRDD, missingDataColumn: Int)

  def handleMissingData(record: RowRecord): String
}
